pkg/app/piped/cloudprovider/kubernetes/state.go

```diff
@@ -469,7 +470,6 @@ func determineServiceHealth(obj *unstructured.Unstructured) (status model.Kubern
 	if len(s.Status.LoadBalancer.Ingress) == 0 {
 		status = model.KubernetesResourceState_OTHER
 		desc = "Ingress points for the load-balancer are in progress"
-		return
```

Comment:
You're correct. In the current implementation, it's definitely an unneeded statement. But it explicitly shows us it must return here even if the additional check is appended following this in the future.

Reply:
I see 👍 
I'll revert this change.

Reply:
Sorry I am not sure what you mean. What are you stuck with?

Reply:
You think about future additional checks, right? For example:
```go
...
     if len(s.Status.LoadBalancer.Ingress) == 0 {
		status = model.KubernetesResourceState_OTHER
		desc = "Ingress points for the load-balancer are in progress"
                 return
      }
      if express {
          status = ...
          desc = ...
          return
     }
      return
}

Reply:
My opinion is that `if len(s.Status.LoadBalancer.Ingress) == 0 {...` should return if we have additional check under it.

Reply:
I think we don't need `return` now because there is no check under `if len(s.Status.LoadBalancer.Ingress) == 0 {...`.

Reply:
>My opinion is that if len(s.Status.LoadBalancer.Ingress) == 0 {... should return if we have additional check under it.

We're on the same page.

>I think we don't need return now because there is no check under if len(s.Status.LoadBalancer.Ingress) == 0 {....

I feel you. Currently the result is the same whether it is there or not, but I think explicitly indicating it will prevent us from forgetting to add a return statement here in the future.

I'd say it is more important to reduce that risk as much as possible than to shorten just one line, but what do you think?

Reply:
Thank you for explaining in detail!
It does make sense to me. I'll revert this change.